### PR TITLE
Add sentiment to query pipeline

### PR DESCRIPTION
**Pull request by chethus**
Include sentiment analysis in results as shown in demo:
image

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -60,7 +60,7 @@
         <artifactId>appengine-maven-plugin</artifactId>
         <version>2.2.0</version>
         <configuration>
-          <deploy.projectId>PROJECT_ID_HERE</deploy.projectId>
+          <deploy.projectId>cbhateja-step-2020</deploy.projectId>
```
Comment (QuinnDACollins) [inline portfolio/pom.xml]: can we keep this as PROJECT_ID_HERE

File: portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java [hunk 2]
```diff
@@ -27,10 +28,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         String captions = request.getParameter(PropertyNames.CAPTIONS);
         Video dummyVideo = new Video(videoId, title, description, captions);
         
-        // Calculate the query.
+        // Calculate the query and sentiment.
         Query query = new Query(dummyVideo, QUERY_SIZE);
+        float sentimentScore = new SentimentTools(dummyVideo).getScore();
```
Comment (QuinnDACollins) [inline portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java:34]: NIT: Whitespace

File: portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
+import com.google.sps.data.SentimentTools;
+import com.google.sps.data.Video;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
```
Comment (QuinnDACollins) [inline portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java:14]: NIT: Whitespace

File: portfolio/src/main/webapp/results.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+// Set the title of the page and sentiment based on parameters when the page loads.
+function initialize() {
+    // Set title text and display title.
+    const titleText = "Analysis of " + urlParams.get(paramNames.videoTitle);
+    const titleElement = document.getElementById("analysis-title");
+    titleElement.textContent = titleText;
+    titleElement.style.display = "block";
+
```
Comment (QuinnDACollins) [inline portfolio/src/main/webapp/results.js:11]: NIT: Whitespace

**Review by QuinnDACollins — APPROVED** [review]
Other than the pom.xml change and some whitespace, LGTM.

File: portfolio/pom.xml (hunk 1 above)
Comment (chethus) [reply portfolio/pom.xml]: Yeah it's changed back in the second commit. I'll try to keep it in all commits in the future.

File: portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java (hunk 2 above)
Comment (chethus) [reply portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java:34]: In this case I was trying to use whitespace intentionally to separate different steps. Should I avoid this in the future? Hoping Nikhil can chime in as well.

Comment (nikhilbirmiwal) [reply portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java:34]: Hey. I think the way the code looks now is good.

File: portfolio/src/main/webapp/results.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Extract parameters from the URL (may use a Servlet to fetch details later).
+const urlParams = new URLSearchParams(window.location.search);
+const paramNames = {
```
Comment (nikhilbirmiwal) [inline portfolio/src/main/webapp/results.js]: Thought: Should this be a in a constants JS file?

**Review by nikhilbirmiwal — APPROVED** [review]

File: portfolio/src/main/webapp/results.js (hunk 5 above)
Comment (chethus) [reply portfolio/src/main/webapp/results.js]: Added, thanks for the suggestion!

**Pull request merged**